python/ray/train/lightning/lightning_checkpoint.py

```diff
@@ -0,0 +1,60 @@
+from typing import TYPE_CHECKING, Optional, Type, Union, Dict
+
+from ray.air.checkpoint import Checkpoint
+from ray.util.annotations import PublicAPI
+
+if TYPE_CHECKING:
+    from pytorch_lightning import LightningModule
+    from ray.data.preprocessor import Preprocessor
+
+
+@PublicAPI(stability="alpha")
+class LightningCheckpoint(Checkpoint):
+    """A ``ray.air.checkpoint.Checkpoint`` with Lightning-specific functionality.
+
+    Create this from a generic :py:class:`~ray.air.checkpoint.Checkpoint` by calling
+    ``LightningCheckpoint.from_checkpoint(ckpt)``.
+    """
+
+    @classmethod
+    def from_model(
+        cls,
+        model: "LightningModule",
+        model_params: Dict,
+        *,
+        preprocessor: Optional["Preprocessor"] = None,
+    ) -> "LightningCheckpoint":
+        """Create a :py:class:`~ray.air.checkpoint.Checkpoint` that stores a Lightning
+        module.
+
+        Args:
+            model: The LightningModule to store in the checkpoint.
+            model_params: The kwargs to be passed to ``model``'s initialization.
```

Comment:
so that we can store both the instantiated model's state dict and the params used to initialize the model (which is useful so we can re-instantiate the model with proper initialization params and load the state dict in `get_model`)